### ISSUE TITLE
Wrong detection in javascript:S3854 (super called twice)

### ISSUE THREAD

**Issue by AMorgaut** [SonarSource/SonarJS#1357]
**I want to report a bug.**

**SonarJS version:** 5.2.1<br>
**SonarQube version:** Community Edition Version 7.8 (build 26217)<br>
**SonarLint version:** 4.1.1.3345<br>

**Rule key:** javascript:S3854

**Reproducer** 
```js
class Collection extends Array {
    constructor(arg) {
        if (typeof arg === 'number') {
            super(arg);
        } else {
            super();  // False-positive here

            if (typeof arg[Symbol.iterator] === 'function') {
                this.push(...arg);
            }
        }
    }
}
```

**Expected behavior**

Might not be the cleanest pattern but here this rule should not raise an error as **`super()` will always be called only once**, and then the rules is respected

**Comment by yassin-kammoun-sonarsource** [comment]
Fixed since #2279.
